src/test/kotlin/com/terraformation/backend/tracking/scenario/ExpectedSiteResultStep.kt

```diff
@@ -0,0 +1,132 @@
+        "Site",
+        scenario,
+        actualResult,
+        survivalRate,
+        baseline,
+    ) {
+  val strata = mutableListOf<Stratum>()
+
+  fun stratum(number: Int, survivalRate: Int? = null, init: Stratum.() -> Unit) =
+      initAndAppend(
+          Stratum(
+              number,
+              survivalRate,
+              baseline?.strata?.firstOrNull { it.number == number },
+          ),
+          strata,
+          init,
+      )
+
+  override fun finish() {
+    if (baseline != null) {
+      mergeBaselineRates()
+
+      val stratumNumbers = strata.map { it.number }.toSet()
+      baseline.strata.filter { it.number !in stratumNumbers }.forEach { stratum(it.number) {} }
+    }
+  }
+
+  inner class Stratum(val number: Int, survivalRate: Int? = null, baseline: Stratum?) :
+      ExpectedObservationResult<ObservationStratumResultsModel, Stratum>(
+          "Stratum $number",
+          scenario,
+          actualResult.strata.first { it.name == "$number" },
+          survivalRate,
+          baseline,
+      ) {
+    val substrata = mutableListOf<Substratum>()
+
+    fun substratum(number: Int, survivalRate: Int? = null, init: Substratum.() -> Unit) =
+        initAndAppend(
+            Substratum(
+                number,
+                survivalRate,
+                baseline?.substrata?.firstOrNull { it.number == number },
+            ),
+            substrata,
+            init,
+        )
+
+    override fun finish() {
+      if (baseline != null) {
+        mergeBaselineRates()
+
+        val ourSubstratumNumbers = substrata.map { it.number }.toSet()
+        baseline.substrata
+            .filter { it.number !in ourSubstratumNumbers }
+            .forEach { substratum(it.number) {} }
+      }
+    }
+
+    inner class Substratum(val number: Int, survivalRate: Int?, baseline: Substratum?) :
+        ExpectedObservationResult<ObservationSubstratumResultsModel, Substratum>(
+            "$name substratum $number",
+            scenario,
+            actualResult.substrata.first { it.name == "$number" },
+            survivalRate,
+            baseline,
+        ) {
+      val plots = mutableListOf<Plot>()
+
+      fun plot(number: Long, survivalRate: Int? = null, init: Plot.() -> Unit) =
+          initAndAppend(
+              Plot(
+                  number,
+                  resultForPlot(number),
+                  survivalRate,
+                  baseline?.plots?.firstOrNull { it.number == number },
+              ),
+              plots,
+              init,
+          )
+
+      override fun finish() {
+        if (baseline != null) {
+          mergeBaselineRates()
+
+          val ourPlotNumbers = plots.map { it.number }.toSet()
+          baseline.plots.filter { it.number !in ourPlotNumbers }.forEach { plot(it.number) {} }
+        }
+      }
+
+      private fun resultForPlot(number: Long) =
+          actualResult.monitoringPlots.firstOrNull { it.monitoringPlotNumber == number }
```

Comment:
Not sure if this is the right place for this, but can we add a kdoc for this function? This one seems to be the only one that wasn't clear from the name (for me at least). Doesn't have to be in this PR if preferred